---
title: "Bacterial isolate - Whole genome sequencing report" 
date: "`r Sys.Date()`"
output: html_document
params:
  csv : ""
---
```{css, echo=FALSE}
.table caption {
    color: darkblue;
    font-weight: bold;
}
```
```{css, echo = FALSE}
  .nav-pills>li>a:hover, .nav-pills>li>a:focus, .nav-pills>li.active>a,     .nav-pills>li.active>a:hover, .nav-pills>li.active>a:focus{
     background-color: darkblue;
     }
```


<br>

### AMR,VF and BUSCO RESULTS {.tabset .tabset-pills}
```{r, results='asis',echo=FALSE}

  samplelist <- read.csv(params$csv,header=FALSE,sep = ',')[-1,]
  sampleid <- samplelist[,c(1)]
  for (i in sampleid){

    cat("\n")
    cat("####", i, "\n")

    rmlst_id <- (paste(i,"_speciesid.txt",sep=""))
    rmlst_file <- readLines(rmlst_id)
    # Display the content
    cat("```\n")
    cat(rmlst_file, sep = "\n")
    cat("\n```\n")
    
  
  
    vf_id <- (paste(i,"_vf.csv",sep=""))
    vf_raw <- read.csv(vf_id,header = TRUE,sep="\t")
    vf_final <- vf_raw[,c(1,6,10,11,15)]
    colnames(vf_final) <- c("SAMPLE","GENE","%COVERAGE","%IDENTITY","VIRULENCE FACTOR")
    print(knitr::kable(vf_final,align= 'lcccl',caption = "Virulence factors - \n Predicted based on https://doi.org/10.1186/s12866-016-0782-8 using abricate tool"))
    cat("\n")

    amr_id <- (paste(i,"_AMR.csv",sep=""))
    amr_raw <- read.csv( amr_id,header = TRUE,sep="\t")
    amr_final <- amr_raw[,c(1,6,10,11,15)]
    colnames(amr_final) <- c("SAMPLE","GENE","%COVERAGE","%IDENTITY","AMR GENE")
    print(knitr::kable(amr_final,align= 'lcccl',caption = "AMR - \n Predicted based using abricate tool and CARD database (https://card.mcmaster.ca/)"))
    cat("\n")

    
    busco_id <- (paste(i,"_busco.txt",sep=""))
    busco_file <- readLines(busco_id)
    # Display the content
    cat("```\n")
    cat(busco_file, sep = "\n")
    cat("\n```\n")

    flyeinfo_id <- (paste(i,"_flye-info.txt",sep=""))
    flyeinfo_raw <- read.csv( flyeinfo_id,header = TRUE,sep="\t")
    flyeinfo_final <- flyeinfo_raw[,c(1,2,3,4)]
    colnames(flyeinfo_final) <- c("SEQUENCE", "LENGTH", "COVERAGE", "IS_CIRCULAR")

    # Format the LENGTH column with appropriate units (bp, Kb, Mb, Gb)
    flyeinfo_final$LENGTH <- sapply(flyeinfo_final$LENGTH, function(x) {
      if (x >= 1e9) {
        paste(round(x / 1e9, 2), "Gb")
      } else if (x >= 1e6) {
        paste(round(x / 1e6, 2), "Mb")
      } else if (x >= 1e3) {
        paste(round(x / 1e3, 2), "Kb")
      } else {
        paste(x, "bp")
      }
    })
    
    # Display the flyeinfo table with formatted lengths
    
    print(knitr::kable(flyeinfo_final,align= 'llll',caption = "Assembly statistics - \n Generated using flye assembler within dragonflye (https://github.com/rpetit3/dragonflye)"))
  
    
    


    }


```

<br>
